### ISSUE TITLE
Create concept of states for Clubs: pending, approved, denied

### ISSUE THREAD

**Issue by hannahkane** [mozilla/teach.webmaker.org#921]
* Pending: does not display publicly, viewable/editable by user (will need to provide entry point and view)
 * Approved: displayed publicly, editable by user
 * Denied: does not display publicly, not viewable/editable by user (could be deleted from dbase actually, after a week or something)

New clubs submissions should be Pending by default

**Comment by toolness** [comment]
One very easy way to show clubs that are pending or denied is to have them display *only* when the club owner is logged-in, and provide a UI affordance indicating this (e.g. a big "PENDING APPROVAL" label in yellow, or "APPROVAL DENIED" in red, next to the club name). This way we don't need to add a whole new UI.

Note also that there's already a concept of "visible" in the admin UI:

![2015-05-18_6-52-42](https://cloud.githubusercontent.com/assets/124687/7679428/966fd7fa-fd2a-11e4-9e95-b79774aa3e2c.jpg)

If we just add a "status" to the model then it can essentially indicate the reason for the club being active vs. not-active.

**Comment by hannahkane** [comment]
release this initially with default state approved; later, once front-end work is done, change default state to pending

**Comment by toolness** [comment]
Er, I have a question...

If a club's status is already "approved" or "denied" and a user makes an edit to it, should the status be reset to "pending", or should it remain unchanged? Or should it be dependent on the existing status, e.g. denied clubs get set to "pending" after edits are made but approved clubs remain approved?

**Comment by toolness** [comment]
Also, is my [comment above](https://github.com/mozilla/teach.webmaker.org/issues/921#issuecomment-103015139) an OK assumption to have? Here it is for clarity:

> One very easy way to show clubs that are pending or denied is to have them display only when the club owner is logged-in, and provide a UI affordance indicating this (e.g. a big "PENDING APPROVAL" label in yellow, or "APPROVAL DENIED" in red, next to the club name). This way we don't need to add a whole new UI.

**Comment by toolness** [comment]
Update from today's stand:

* Having a user access their pending or denied club from the map/list view is ok for now, but we'll eventually want some sort of "admin panel" UX for users (not to be confused w/ the admin panel for staff on the teach-api site).
* We're not assuming that our vetted users are malicious, so if an approved club has been edited, it stays approved.
* If a denied club is edited, its status should change to pending, since it's assumed that the user has just fixed whatever caused it to be denied.

**Comment by toolness** [comment]
Ok, I've implemented the back-end logic, as it's been described so far, in https://github.com/mozilla/teach-api/pull/16.

Some notes:
* We're currently *only* emailing anyone (the club owner and teach staff) when the club has been created. We're not emailing the club owner when their club has been approved or denied through the admin UI. If this is intentional, I assume it's because the teach staff/RC will be emailing them manually with more details on the rationale of the approve/deny?
* We're also not emailing teach staff when a club owner edits their denied club and the club status is set back to pending. Not sure if we should pro-actively do this, or assume that the teach staff will already be watching for the change, and/or assume that the club owner will manually alert the teach staff through email.

Let me know if you want to change any of those things, otherwise I think the backend is ready.

Note though that I realized it's actually easier to **not** do what I suggested in the stand [2 days ago](https://github.com/mozilla/teach.webmaker.org/issues/921#issuecomment-105992321), to "release this initially with default state approved"... which means that we could deploy to staging in one of two ways:

1. Land the backend changes to staging before landing the front-end changes (which haven't been implemented yet). This will actually leave the staging site with an unintentional [shadowban](http://www.urbandictionary.com/define.php?term=shadowban)-esque policy, where the front-end will be telling users that their clubs are on the map, and they will appear to the user to be published, but they will not actually be visible to the rest of the world because they're regarded as "pending" by the backend.
2. Wait until the front-end changes are implemented, and then land both backend and front-end changes simultaneously.

Obviously I think we shouldn't land anything in production until *everything* is ready, but let me know if you disagree. :)

**Comment by toolness** [comment]
Oops scratch that, I still have to implement "Change success message copy from "We've added your Club!" to "Thanks for your submission! We'll review it and be in touch shortly." in the back-end, as mentioned in #920. Will do that ASAP.

**Comment by toolness** [comment]
Whoops, nevermind, that message appears on the front-end, not in the transactional email. I should note, though, that the transactional email sent to the club creator is *not* currently changed... I guess its copy still makes sense as part of the new approval flow, though?

```python
"""
%(username)s,

You did it! You put your city on the map and showed your dedication to increasing web literacy around the world. This is a big first step, so thank you for all that you are doing to teach the web.

Whether you are a new club leader or an expert at running clubs, we recommend you get started by meeting others in our community. Join the conversation in our discussion forum [http://discourse.webmaker.org/category/clubs], or tell us what you're teaching on Twitter using hashtag #teachtheweb.

A member of our team will send you an email soon to see how you're doing and make sure your club is on the road to success!

Michelle Thorne,
on behalf of the Mozilla Learning team

P.S. Be sure to check out our tips for running clubs at %(TEACH_SITE_URL)s/clubs/ !
"""
```

**Comment by toolness** [comment]
In regards to [my above comment](https://github.com/mozilla/teach.webmaker.org/issues/921#issuecomment-106767432), I think landing the back-end changes on staging *before* the front-end changes is actually the best option, even though it will temporarily result in weird shadowbanny behavior. This is because when mavis reviews my PRs for the front-end changes, she should be able to easily test them against the new behavior of the back-end, preferably without having to manually set up the back-end on her development machine.

**Comment by hannahkane** [comment]
Thanks, @toolness. As discussed just now in chat - yes, let's release the back end to staging, and stick with no auto-emails ('cept the initial one) for now. This allows us to iterate on the process (and the email copy!) until we get it right.

**Comment by toolness** [comment]
Ok, just pushed the back end to staging!

**Comment by toolness** [comment]
Since this has been merged into teach-api, I'm closing this issue.
